package defpackage;

import kotlin.jvm.functions.Function3;
import kotlin.jvm.internal.Ref;

/* renamed from: pv5  reason: default package */
public final class pv5 implements ds5 {
    public final /* synthetic */ Ref.ObjectRef a;
    public final /* synthetic */ Function3 b;
    public final /* synthetic */ ds5 c;

    public pv5(Ref.ObjectRef objectRef, Function3 function3, ds5 ds5) {
        this.a = objectRef;
        this.b = function3;
        this.c = ds5;
    }

    /* JADX WARNING: Removed duplicated region for block: B:14:0x003c  */
    /* JADX WARNING: Removed duplicated region for block: B:20:0x006b A[RETURN] */
    /* JADX WARNING: Removed duplicated region for block: B:8:0x0024  */
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public final java.lang.Object a(java.lang.Object r7, kotlin.coroutines.Continuation r8) {
        /*
            r6 = this;
            boolean r0 = r8 instanceof defpackage.ov5
            if (r0 == 0) goto L_0x0013
            r0 = r8
            ov5 r0 = (defpackage.ov5) r0
            int r1 = r0.v
            r2 = -2147483648(0xffffffff80000000, float:-0.0)
            r3 = r1 & r2
            if (r3 == 0) goto L_0x0013
            int r1 = r1 - r2
            r0.v = r1
            goto L_0x0018
        L_0x0013:
            ov5 r0 = new ov5
            r0.<init>(r6, r8)
        L_0x0018:
            java.lang.Object r8 = r0.c
            java.lang.Object r1 = kotlin.coroutines.intrinsics.IntrinsicsKt.getCOROUTINE_SUSPENDED()
            int r2 = r0.v
            r3 = 2
            r4 = 1
            if (r2 == 0) goto L_0x003c
            if (r2 == r4) goto L_0x0034
            if (r2 != r3) goto L_0x002c
            kotlin.ResultKt.throwOnFailure(r8)
            goto L_0x006c
        L_0x002c:
            java.lang.IllegalStateException r6 = new java.lang.IllegalStateException
            java.lang.String r7 = "call to 'resume' before 'invoke' with coroutine"
            r6.<init>(r7)
            throw r6
        L_0x0034:
            kotlin.jvm.internal.Ref$ObjectRef r6 = r0.b
            pv5 r7 = r0.a
            kotlin.ResultKt.throwOnFailure(r8)
            goto L_0x0056
        L_0x003c:
            kotlin.ResultKt.throwOnFailure(r8)
            kotlin.jvm.internal.Ref$ObjectRef r8 = r6.a
            T r2 = r8.element
            r0.a = r6
            r0.b = r8
            r0.v = r4
            kotlin.jvm.functions.Function3 r4 = r6.b
            java.lang.Object r7 = r4.invoke(r2, r7, r0)
            if (r7 != r1) goto L_0x0052
            return r1
        L_0x0052:
            r5 = r7
            r7 = r6
            r6 = r8
            r8 = r5
        L_0x0056:
            r6.element = r8
            ds5 r6 = r7.c
            kotlin.jvm.internal.Ref$ObjectRef r7 = r7.a
            T r7 = r7.element
            r8 = 0
            r0.a = r8
            r0.b = r8
            r0.v = r3
            java.lang.Object r6 = r6.a(r7, r0)
            if (r6 != r1) goto L_0x006c
            return r1
        L_0x006c:
            kotlin.Unit r6 = kotlin.Unit.INSTANCE
            return r6
        */
        throw new UnsupportedOperationException("Method not decompiled: defpackage.pv5.a(java.lang.Object, kotlin.coroutines.Continuation):java.lang.Object");
    }
}
